### ISSUE TITLE
opening a pipe to a `while read` loop flags assignment of `$?` as SC2319

### ISSUE THREAD

**Issue by solstice333** [koalaman/shellcheck#3165]
Min repro:

```bash
#!/usr/bin/env bash

while read line; do
    rev=$line

    [[ "${rev#foo}" == "$rev" ]]
    is_foo=$?   # This $? refers to a condition, not a command. Assign to a variable to avoid it being overwritten. See SC2319.

    echo "$rev: $is_foo"
done < $TEMP/foo.lst | sort
```
where `is_foo=$?` is flagged with SC2319. The expectation is for this line *not* to be flagged bc `$?` is being assigned.

Removing the pipe of `while read ...` to `sort` causes the SC2319 warning to disappear:

```bash
#!/usr/bin/env bash

while read line; do
    rev=$line

    [[ "${rev#foo}" == "$rev" ]]
    is_foo=$?

    echo "$rev: $is_foo"
done < $TEMP/foo.lst
```
